### ISSUE TITLE
Bun crash when using $ in a macro on windows

### ISSUE THREAD

**Issue by matthieuvancayzeele** [oven-sh/bun#12367]
### How can we reproduce the crash?

Just a simple program with no libraries

src/index.ts
`
import { ver } from '../macros/ver' with { type: "macro" };
console.log(ver())
`

macros/ver.ts
`
import { $ } from 'bun'

export async function ver() {
    const v = await $`echo "0.0.1"`.text()
    return v
}
`

program started with 
`
bun .\src\index.ts
`

### Relevant log output

```shell
PS C:\Users\geekg\Desktop\Elysia\app> bun .\src\index.ts
============================================================
Bun v1.1.12 (43f0913c) Windows x64
Args: "C:\Users\geekg\.bun\bin\bun.exe" ".\src\index.ts"
Features: jsc macros(2) shell tsconfig(2) 
Builtins: "bun:main" 
Elapsed: 41ms | User: 0ms | Sys: 0ms
RSS: 103.65MB | Peak: 103.65MB | Commit: 0.15GB | Faults: 25523

panic(main thread): Segmentation fault at address 0x8
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.12/wa143f0913AiggglEg59+2CkyyrIutq8Qw7u8Qu6soKonu3aqq4lY479zTuxygUmx+zTA2AQ
```

### Stack Trace (bun.report)

Bun v1.1.12 ([`43f0913`](<https://github.com/oven-sh/bun/tree/43f0913c3875cbf260c60868268adc5953b09a80>)) on windows x86_64 [AutoCommand]

Segmentation fault at address 0x00000008

- `us_wakeup_loop`
- [`interpreter.zig:322`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L322>): `close`
- [`interpreter.zig:742`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L742>): `deinitImpl`
- [`interpreter.zig:2570`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L2570>): `finish`
- [`interpreter.zig:5461`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L5461>): `done`
- [`interpreter.zig:6831`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L6831>): `start`
- [`interpreter.zig:5171`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L5171>): `callImpl`
- [`interpreter.zig:4495`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L4495>): `transitionToExecStateAndYield`
- [`interpreter.zig:1916`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L1916>): `next`
- [`interpreter.zig:1802`](<https://github.com/oven-sh/bun/blob/43f0913c3875cbf260c60868268adc5953b09a80/src/shell/interpreter.zig#L1802>): `start`

**Comment by matthieuv-chabe** [comment]
Yes exact same issue

**Comment by dylan-conway** [comment]
Fixed in #12780
